src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -167,11 +370,152 @@ public int getBonusVision()
+            BlockPosUtil.writeToNBT(wayPointCompound, TAG_TARGET, pos);
+
+
+            wayPointTagList.appendTag(wayPointCompound);
+        }
+        compound.setTag(TAG_PATROL_TARGETS, wayPointTagList);
+
+        BlockPosUtil.writeToNBT(compound, TAG_GUARD, guardPos);
+    }
+
+    @Override
+    public void serializeToView(@NotNull final ByteBuf buf)
+    {
+        super.serializeToView(buf);
+        buf.writeBoolean(assignManually);
+        buf.writeBoolean(retrieveOnLowHealth);
+        buf.writeBoolean(patrolManually);
+        buf.writeInt(task.ordinal());
+        buf.writeInt(job == null ? -1 : job.ordinal());
+        buf.writeInt(patrolTargets.size());
+
+        for(BlockPos pos: patrolTargets)
+        {
+            BlockPosUtil.writeToByteBuf(buf, pos);
+        }
+
+        BlockPosUtil.writeToByteBuf(buf, guardPos);
+    }
+
+    /**
+     * Check if the guard should retrieve on low health.
+     * @return true if so.
+     */
+    public boolean shallRetrieveOnLowHealth()
+    {
+        return retrieveOnLowHealth;
+    }
+
+    /**
+     * Set if the guard should retrieve on low health.
+     * @param retrieveOnLowHealth state to set.
+     */
+    public void setRetrieveOnLowHealth(final boolean retrieveOnLowHealth)
+    {
+        this.retrieveOnLowHealth = retrieveOnLowHealth;
+        this.markDirty();
+    }
+
+    /**
+     * Check if the guard should patrol manually (Use defined values).
+     * @return true if so.
+     */
+    public boolean shallPatrolManually()
+    {
+        return patrolManually;
+    }
+
+    /**
+     * Set if the guard should patrol manually.
+     * @param patrolManually the state to set.
+     */
+    public void setPatrolManually(final boolean patrolManually)
+    {
+        this.patrolManually = patrolManually;
+        this.markDirty();
+    }
+
+    /**
+     * Set the task of the guard.
+     * @param task the task to set.
+     */
+    public void setTask(final Task task)
+    {
+        this.task = task;
+        this.markDirty();
+    }
+
+    /**
+     * Getter of the position the gurd should guard.
+     * @return the blockPos position.
+     */
+    public BlockPos getGuardPos()
+    {
+        return guardPos;
+    }
+
+    /**
+     * Set if the jobs should be assigned manually.
+     * @param assignManually the state to set.
+     */
+    public void setAssignManually(final boolean assignManually)
+    {
+        this.assignManually = assignManually;
+        this.markDirty();
+    }
+
     /**
      * The client view for the baker building.
      */
     public static class View extends AbstractBuildingWorker.View
     {
+        public boolean assignManually = false;
+        public boolean retrieveOnLowHealth = false;
+        public boolean patrolManually = false;
+        public Task task = Task.GUARD;
+        public GuardJob job = null;
+        public BlockPos guardPos = getLocation();
+
+        public ArrayList<BlockPos> patrolTargets = new ArrayList<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "patrolTargets" object should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1319)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make patrolTargets a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)